components/core/CMakeLists.txt

```diff
@@ -337,6 +358,13 @@ target_compile_features(clp
         PRIVATE cxx_std_17
         )
 
+# Generate C++ Kibana Parser
+# This needs to be done using add_subdirectory because antlr's c++ build
+# system can't handle having the grammar (.g4) in a different directory than the
+# CMakeLists.txt file.
+add_subdirectory(src/grammar/kibana)
+include_directories(${ANTLR4_INCLUDE_DIRS} ${GENERATED_KIBANA_INCLUDE_DIR})
```

Comment:
Can this be made to be `target_include_directories`?

In general, we should try to move to the newer target_* commands as much as possible.